tests/unit/lib/test_datachain.py

```diff
@@ -1317,7 +1322,7 @@ def test_to_csv_features_nested(tmp_dir, test_session):
 @pytest.mark.parametrize("object_name", (None, "test_object_name"))
 @pytest.mark.parametrize("model_name", (None, "TestModelNameExploded"))
 def test_explode(tmp_dir, test_session, column_type, object_name, model_name):
-    df = pd.DataFrame(DF_DATA)
+    df = pd.DataFrame(DF_DATA_NESTED_NOT_NORMALIZED)
```

Comment:
Here, we parse a way more complicated JSON (with optional values in JSON, with non-normalized top level values) - that pretty much drove a lot of changes in the code, besides the main one - `schema_sample_size`. `schema_sample_size` didn't work w/o a lot of fixes around